### DISCUSSION TITLE
Custom client prevent stopping Windows service

### DISCUSSION THREAD

**Discussion by D-tack** [rustdesk/rustdesk-server-pro#909] [Feature Request]
Hiya,

I'm planning to use RustDesk as a replacement for TeamViewer but I encountered an issue that might cause me and my team some trouble in the long run.

I have noticed that even though the settings are disabled in the creation of a custom client the customer on site can go to the 3 dots and manually stop the service causing a loss of connection. The result of this is either a painful long distance drive to the client location or (video) calling said customer to guide them to re-enable the service manually.

Please see the screenshots attached.
![image](https://github.com/user-attachments/assets/6099495f-8f33-4790-bc8b-4e63b701a2fc)
![image](https://github.com/user-attachments/assets/bfdfc9e0-e1b6-46af-8a15-201a67308e37)
![image](https://github.com/user-attachments/assets/2b4f49ff-8165-4879-a44e-65c9496fe240)
![image](https://github.com/user-attachments/assets/c4ed29cb-33df-49ba-a633-0bf6f44d1179)

I'm not sure if I missed out on an advanced setting override, if I did please forgive me as I have read the documentation carefully and searched Github issues.

If the setting does not exist yet is it possible to have it added? (I think more people can vouch for this)
Something like prevent-disabling-service=Y
Alternatively secure the button with a pin code if the above is not possible?

**Comment by rustdesk** [comment]
Thanks, we will work out a solution. 

@21pages add `hide-stop-service`.

**Comment by vhewitt29** [comment]
Any movement on this, I had corp. ver. of TeamViewer, and now paid for pro ver. of rust desk looking for same amount of access and control

**Comment by StRosesIT** [comment]
Is there an update for this? I too am replacing TeamViewer and I would like to hide the stop service button from users, or have an option to set a PIN to access any of the settings behind the 3 dots.

**Comment by rustdesk** [comment]
I will add this option `hide-stop-service`. Also disable this from tray menu.

**Comment by rustdesk** [comment]
@StRosesIT but how about if user stop it from the windows service manager?

**Reply by StRosesIT** [reply]
You need admin privileges to stop a service from Services, which my users don't have.

**Comment by rustdesk** [comment]
Another question, user can also stop from command line if he has admin.  `--uninstall-service`

**Reply by StRosesIT** [reply]
My users don't have local administrator rights on their devices so they can't do this.

**Comment by rustdesk** [comment]
The "stop" service button also requires admin. Then what is the point of hide it?

**Reply by StRosesIT** [reply]
Ideally, I would like to hide all the settings behind the 3 dots from users - I will look to raise this as a new feature request.

I do have a handful of special users that have local admin and if we have more in future, I would like to be able to hide that button. It's only 2 clicks for them to stop it through the client. The other routes require more clicks, and I can add settings to hide them better from the user in the OS.

**Reply by 21pages** [reply]
When the service is stopped, should the Start/Stop Service button be displayed?

**Comment by mnunes473** [comment]
Do you already have a solution for this issue?

I'm testing the client configuration to run as unattended access, but the problem of stopping the service is impacting my choice of RustDesk.

I need to keep the application running without interruption, so I need to prevent the user from simply clicking "stop service," which would remove the service and cause further headaches.

In tests, I managed to work around this with a script installation, but it would be more practical to have this already configured in the application. I've already tried adding `hide-stop-service`, but it had no effect; I can still click to stop the service.
